controller/linodemachine_controller.go

```diff
@@ -103,76 +103,31 @@ func (r *LinodeMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	linodeMachine := &infrav1alpha1.LinodeMachine{}
 	if err := r.Client.Get(ctx, req.NamespacedName, linodeMachine); err != nil {
 		if err = client.IgnoreNotFound(err); err != nil {
-			log.Error(err, "Failed to fetch Linode machine")
+			log.Error(err, "Failed to fetch LinodeMachine")
 		}
 
 		return ctrl.Result{}, err
 	}
 
-	machine, err := kutil.GetOwnerMachine(ctx, r.Client, linodeMachine.ObjectMeta)
-	switch {
-	case err != nil:
-		if err = client.IgnoreNotFound(err); err != nil {
-			log.Error(err, "Failed to fetch owner machine")
-		}
-
+	machine, err := r.getOwnerMachine(ctx, *linodeMachine, log)
+	if err != nil || machine == nil {
 		return ctrl.Result{}, err
-	case machine == nil:
-		log.Info("Machine Controller has not yet set OwnerRef, skipping reconciliation")
-
-		return ctrl.Result{}, nil
-	case skippedMachinePhases[machine.Status.Phase]:
-
-		return ctrl.Result{}, nil
-	default:
-		match := false
-		for i := range linodeMachine.OwnerReferences {
-			if match = linodeMachine.OwnerReferences[i].UID == machine.UID; match {
-				break
-			}
-		}
-
-		if !match {
-			log.Info("Failed to find the referenced owner machine, skipping reconciliation", "references", linodeMachine.OwnerReferences, "machine", machine.ObjectMeta)
-
-			return ctrl.Result{}, nil
-		}
 	}
+	log = log.WithValues("LinodeMachine: ", machine.Name)
 
-	log = log.WithValues("Linode machine: ", machine.Name)
-
-	cluster, err := kutil.GetClusterFromMetadata(ctx, r.Client, machine.ObjectMeta)
-	switch {
-	case err != nil:
-		if err = client.IgnoreNotFound(err); err != nil {
-			log.Error(err, "Failed to fetch cluster by label")
-		}
-
-		return ctrl.Result{}, err
-	case cluster == nil:
-		err = errors.New("missing cluster")
-
-		log.Error(err, "Missing cluster")
-
-		return ctrl.Result{}, err
-	case cluster.Spec.InfrastructureRef == nil:
-		err = errors.New("missing infrastructure reference")
-
-		log.Error(err, "Missing infrastructure reference")
-
+	cluster, err := r.getClusterFromMetadata(ctx, *machine, log)
+	if err != nil || cluster == nil {
 		return ctrl.Result{}, err
 	}
 
-	linodeCluster := &infrav1alpha1.LinodeCluster{}
-
 	machineScope, err := scope.NewMachineScope(
 		ctx,
 		r.LinodeApiKey,
 		scope.MachineScopeParams{
 			Client:        r.Client,
 			Cluster:       cluster,
 			Machine:       machine,
-			LinodeCluster: linodeCluster,
+			LinodeCluster: nil,
```

Comment:
For what it's worth, you can simply _not_ specify this field, and it will default to the zero-value for the field's type. Since `LinodeCluster` is a `*T` (pointer to a type), the zero value is `nil`.